llvm/lib/MC/RepoObjectWriter.cpp

```diff
@@ -827,6 +839,43 @@ RepoObjectWriter::writeDebugLineHeader(TransactionType &Transaction,
   return {};
 }
 
+pstore::index::digest
+RepoObjectWriter::updateFragmentDigest(const ticketmd::DigestType &InitailHash,
+                                       FragmentContentsType &FragmentContent) {
+  MD5 FragmentHash;
+  MD5::MD5Result FragmentDigest;
+  auto IsEmptyDependent = FragmentContent.Dependents.empty();
+  if (!IsEmptyDependent) {
+    // If this Fragment has dependents and the dependents have an internal
+    // linkage type, accumulate the dependents' hash to this fragment.
+
+    FragmentHash.update(InitailHash.Bytes);
+    for (auto Dependent : FragmentContent.Dependents) {
+      auto DependentCompilationMember =
+          CompilationMembers[Dependent.absolute()];
+      if (DependentCompilationMember.linkage ==
+          pstore::repo::linkage_type::internal) {
+        FragmentHash.update(
+            makeByteArrayRef(DependentCompilationMember.digest));
+        FragmentHash.update(makeByteArrayRef(DependentCompilationMember.name));
+      }
+    }
+    FragmentHash.final(FragmentDigest);
+  }
+
+  auto const Key =
```

Comment:
If FragmentHash is unconditionally initialized to InitialHash, can this condition be removed?

Reply:
Good point, Fixed and removed this condition.